### ISSUE TITLE
Workflow marking store type 

### ISSUE THREAD

**Issue by klimenttoshkov** [zerodahero/laravel-workflow#36]
UPDATE: My questions are:
1. How do we use state_machine and do we need it at all?
2. Should we stick to Symfony recommendation to avoid setting marking store type?
-- please read below full text

According to the documentation of Symfony [https://symfony.com/doc/current/workflow.html#storing-metadata](url) default Workflow marking store type is inferred based on workflow type. If it is workflow, marking store should be "multiple_state". Their recommendation is to not configure the type manually:

>The marking store type could be “multiple_state” or “single_state”. A single state marking store does not support a model being on multiple places at the same time. This means a “workflow” must use a “multiple_state” marking store and a “state_machine” must use a “single_state” marking store. Symfony configures the marking store according to the “type” by default, so it’s preferable to not configure it.
>
>A single state marking store uses a string to store the data. A multiple state marking store uses an array to store the data.

>The marking_store.type (the default value depends on the type value) and property (default value ['marking']) attributes of the marking_store option are optional. If omitted, their default values will be used. It’s highly recommended to use the default value.

However based on my testing I have found out that default type is "single_state" even when using type workflow (btw. state_machine dumps the workflow with transitions in squares, not with labels above the arrow which makes me believe it is workflow, not state_machine). I have to explicitly set `marking: multiple_state` if I need to support multiple places and avoid errors. At the same time `type: state_machine|workflow` appears to be ignored at all.

```
    'test' => [
        'type' => 'workflow',

        'supports' => [
            Test::class,
        ],

        'places' => [
            'a', 'b',  'c',
            'a1', 'b1', 'b2',
            'a2', 'c1', 'c2',
        ],

        'initial_places' => [
            'a', 'a1', 'a2'
        ],

        'transitions' => [
            't1' => [
                'from' => 'a',
                'to' => 'b',
                'metadata' => [
                    'priority' => 0.5,
                ],
            ],
            't2' => [
                'from' => 'b',
                'to' => 'c'
            ],

            't11' => [
                'from' => 'a1',
                'to' => 'b1'
            ],
            't21' => [
                'from' => 'b1',
                'to' => 'c1'
            ],

            't12' => [
                'from' => 'a2',
                'to' => 'b2'
            ],
            't22' => [
                'from' => 'b2',
                'to' => 'c2'
            ],

        ],

```
Results:
```
   Symfony\Component\Workflow\Exception\NotEnabledTransitionException 

  Transition "t11" is not enabled for workflow "test".
```

And this configuration:
```
    'test' => [
        'type' => 'state_machine', /// NOTE THIS ?
        'marking_store' => [
            'type' => 'multiple_state',
        ],

        'supports' => [
            Test::class,
        ],

        'places' => [
            'a', 'b',  'c',
            'a1', 'b1', 'b2',
            'a2', 'c1', 'c2',
        ],

        'initial_places' => [
            'a', 'a1', 'a2'
        ],
        'transitions' => [
            't1' => [
                'from' => 'a',
                'to' => 'b',
                'metadata' => [
                    'priority' => 0.5,
                ],
            ],
            't2' => [
                'from' => 'b',
                'to' => 'c'
            ],

            't11' => [
                'from' => 'a1',
                'to' => 'b1'
            ],
            't21' => [
                'from' => 'b1',
                'to' => 'c1'
            ],

            't12' => [
                'from' => 'a2',
                'to' => 'b2'
            ],
            't22' => [
                'from' => 'b2',
                'to' => 'c2'
            ],

        ],
```
works fine and moves the workflow into expected places.

**Comment by zerodahero** [comment]
This should be handled in the recent changes to develop as well. I've updated the README as well, please let me know if anything's unclear.

I left a slightly more descriptive comment on the other issue: https://github.com/zerodahero/laravel-workflow/issues/34#issuecomment-766942524

**Comment by zerodahero** [comment]
Addressed in v3.2.1 (see issue: https://github.com/zerodahero/laravel-workflow/issues/34 for more of the conversation, if anyone missed it), closing. Feel free to reopen if there's anything we missed!
